### DISCUSSION TITLE
How does Advanced initialization export authOptions to the front end

### DISCUSSION THREAD

**Discussion by PaulNext** [nextauthjs/next-auth#5069] [Help]
### Question 💬

How to export authOptions to the front-end page when customizing the nextauth route？
The main options are implemented through the API routing function, but the front-end page also needs authOptions. How to export and pass this object to the front-end?

### How to reproduce ☕️

1. Advanced initialization Code

import type { NextApiRequest, NextApiResponse } from "next"
import NextAuth from "next-auth"

export default async function auth(req: NextApiRequest, res: NextApiResponse) {
  // Do whatever you want here, before the request is passed down to `NextAuth`
  return await NextAuth(req, res, {
    ...
  })
}

2. Default authOptions Code
export const authOptions: NextAuthOptions = {
session: {},
providers:[],
...
}

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by saby22** [comment]
Same question here. Any answers would be appreciated.

**Comment by hussamkhatib** [comment]
Been almost a year, did any one of you figured it out ?

**Comment by rexfordessilfie** [comment]
Here you go! Just answered this elsewhere. Let me know if it works for you: https://github.com/nextauthjs/next-auth/discussions/6912#discussioncomment-6025442
